domain: handle restarts attribute correct

ahhud opened the pull request:
restarts attribute for servers and executeble are incremented when they are restarted for any reason.

Resolves #277

ahhud committed:
domain: handle restarts attribute correct

restarts attribute for servers and executeble are incremented when they
are restarted for any reason.

Resolves #277

lazan commented on middleware/domain/source/manager/handle.cpp:
```diff
@@ -601,10 +601,19 @@ namespace casual
+                              handle::scale::instances( state, *executable);
+                              if( spawnable)
+                                 executable->restarts++;
+                           };
```
* What if scale::instances fails? 
* What if there are more than one _spawnable_? 

It's really unfortunate that "we" picked _restarts_... It gets way more complicated then one would like...

Should we rethink this?

lazan replied:
What if we keep track of deaths/exits? And then calculated the restarts for the cli? No that wont work...

lazan replied:
I've been looking for a less "complex" way... The only thing that seems reasonable is to increment `restarts` in  `state::Server::remove( common::strong::process::id pid)` and  `state::Executable::remove( strong::process::id pid)`.  That's when we go from `running` to  `scale_out` via _restart_ .  The meaning of _restarts_ becomes more like _scheduled_restarts_, but to me it feels more sane. 
What do you think @ahhud

lazan commented on middleware/domain/unittest/source/manager/test_manager.cpp:
```diff
@@ -1672,5 +1672,76 @@ namespace casual
+         alias: sleep
+         arguments: [60]
+         instances: 1
+         restart: true
```
Nice. Clever with sleep, even if we have a bug it will only "hang" 60s.

ahhud committed:
domain: handle restarts attribute correct

Changes after pull request review
Changing state attribute to initiated_restarts.
Keeping attribute restarts in public model for now.

Resolves #277

§

lazan commented on middleware/domain/include/domain/manager/state.h:
```diff
@@ -124,7 +123,7 @@ namespace casual
                CASUAL_SERIALIZE( environment);
                CASUAL_SERIALIZE( restart);
-               CASUAL_SERIALIZE( restarts);
+               CASUAL_SERIALIZE( initiated_restarts);
```
Good name.

lazan commented on middleware/domain/source/manager/admin/cli.cpp:
```diff
@@ -348,7 +348,7 @@ namespace casual
       if instances should be restarted if exited
    #r:
-      number of times instances has been restarted
+      number of times instances has been initiated for a restart
```
Nice!

lazan reviewed (approved):
Feels like this makes sense. Nice.
